src/components/Form.js

```diff
@@ -1,16 +1,81 @@
-import React from 'react';
+import './form.css';
+import React, { useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { v4 as uuidc4 } from 'uuid';
+import { addBook } from '../redux/books/books';
+
+const Form = () => {
+  const [title, setTitle] = useState('');
+  const [category, setCategory] = useState('');
+  const [author, setAuthor] = useState('');
+
+  const dispatch = useDispatch();
+
+  const handleSubmit = (e) => {
+    e.preventDefault();
+
+    const book = {
+      item_id: uuidc4(),
+      title: `${title.trim()} , ${author.trim()}`,
+      category,
+    };
+
+    fetch(
+      'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/hU7hRqGMmMkUTHelAE4I/books',
+      {
+        method: 'POST',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify(book),
+      },
+    );
+
+    dispatch(addBook(book));
+
+    setTitle('');
+    setCategory('');
+    setAuthor('');
+  };
 
-const Form = (props) => {
-  // eslint-disable-next-line react/prop-types
-  const { addBook, title, category } = props;
   return (
-    <form className="book-container">
-      <input type="text" placeholder="title" ref={title} />
-      <input type="text" placeholder="auther" ref={category} />
-      <button type="submit" onClick={addBook}>
-        submit
-      </button>
-    </form>
+    <section className="form" onSubmit={handleSubmit}>
+      <h3>ADD NEW BOOK</h3>
```

Comment:
- [ ] What do you think of moving this header to the left as in the original design?

Reply:
All Done